src/addons/build.js

```diff
@@ -62,6 +62,30 @@ module.exports = function(XRegExp) {
     }
 
     /**
+     * Provides a tag function for building regexes using template literals [1]. See GitHub issue
+     * 103 for discussion [2].
+     *
+     * [1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_template_literals
+     * [2]: https://github.com/slevithan/xregexp/issues/103
+     */
+    XRegExp.tag = function (flags) {
+        return function tag (literals /*, ...substitutions */) {
+            var substitutions = [].slice.call(arguments, 1);
+            var subpatterns = substitutions.concat('').map(interpolate);
+            var pattern = literals.raw.map(embedSubpatternAfter).join('');
+            return XRegExp.build(pattern, subpatterns, flags);
+        };
+
+        function interpolate (substitution) {
```

Comment:
Please move `interpolate` and `embedSubpatternAfter` out of and above `XRegExp.tag`, to the same level as `asXRegExp`. And remove the spaces after the function names. :)